File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:

        # Approach - Go from right to left and keep updating the max value
        n = len(arr)
        res = [0] * n

        right_max = -1

        for i in range(n-1, -1, -1):
            res[i] = right_max
            right_max = max(arr[i], right_max)
        
        return res

File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    let n = arr.length;
    let res = new Array(n);
    let right_max = -1;

    for (let i=n-1; i>=0; i--){
        res[i] = right_max;
        right_max = Math.max(right_max, arr[i]);
    }

    return res;
};